internal/pkg/config/watcher/watcher.go

```diff
@@ -100,7 +100,7 @@ func validate(c *K8sConfigMapWatcher) error {
 // Watch watches for events impacting watched ConfigMaps and emits their events across a channel
 func (c *K8sConfigMapWatcher) Watch(ctx context.Context, notifyMe chan<- interface{}) error {
 	glog.V(3).Infof("Watching for ConfigMaps for changes on namespace=%s with labels=%v", c.Namespace, c.ConfigMapLabels)
-	watcher, err := c.client.ConfigMaps(c.Namespace).Watch(metav1.ListOptions{
+	watcher, err := c.client.ConfigMaps(c.Namespace).Watch(context.Background(), metav1.ListOptions{
```

Comment:
this should probably thread the `ctx` passed into this func instead of using a background context. Thoughts?